addons/account_update_tax_tags/wizard/account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,232 @@
+                ON
+                    tax_to_rep_line.repartition_type = 'base'
+                    AND (
+                        -- invoice type doc
+                        (COALESCE(move_caba.move_type, move.move_type) IN ('in_invoice','out_invoice', 'in_receipt', 'out_receipt') AND tax_to_rep_line.invoice_tax_id = tax.id)
+                        OR
+                        -- refund type doc
+                        (COALESCE(move_caba.move_type, move.move_type) IN ('in_refund','out_refund') AND tax_to_rep_line.refund_tax_id = tax.id) 
+                        OR
+                        -- entry type doc: depends on the tax type
+                        -- for base line:
+                        -- balance < 0 and sale tax --> invoice
+                        -- balance > 0 and sale tax --> refund
+                        -- balance > 0 and purchase tax --> invoice
+                        -- balance < 0 and purchase tax --> refund
+                        -- impossible to decide for balance at 0 --> invoice by default
+                        (
+                            move.move_type = 'entry' 
+                            AND (
+                                (
+                                    COALESCE(parent_tax.type_tax_use, tax.type_tax_use) = 'sale'
+                                    AND (
+                                        aml.balance <= 0 AND tax_to_rep_line.invoice_tax_id = tax.id
+                                        OR aml.balance > 0 AND tax_to_rep_line.refund_tax_id = tax.id 
+                                    )
+                                )
+                                OR
+                                (
+                                    COALESCE(parent_tax.type_tax_use, tax.type_tax_use) = 'purchase'
+                                    AND (
+                                        aml.balance >= 0 AND tax_to_rep_line.invoice_tax_id = tax.id
+                                        OR aml.balance < 0 AND tax_to_rep_line.refund_tax_id = tax.id
+                                    )
+                                )
+                            )
+                        ) 
+                    )
+                -- LEFT to allow keeping all aml even those without relation. The NULL will symbolize no relation
+                LEFT JOIN account_account_tag_account_tax_repartition_line_rel rep_tags 
+                ON rep_tags.account_tax_repartition_line_id = tax_to_rep_line.id
+            ), 
+            
+            -- 1.b) Handle tax line: relation aml <-> tag, if no relation, tag is NULL
+            tax_aml_id_rep_tag_to_insert AS (
+                SELECT aml.id AS aml_id, rep_tags.account_account_tag_id AS tag_id
+                FROM account_move_line aml
+                JOIN account_move move ON aml.move_id = move.id AND move.company_id = %(company_id)s AND aml.date >= %(date_from)s
```

Comment:
We don't need that join on account_move. 

You can put a condition in a WHERE to checl aml.company_id = %(company_id)s

Reply:
good catch indeed :+1: